medical/models.py

```diff
@@ -149,13 +168,44 @@ class Service(VersionedModel, ItemOrService):
 
     # validity_from = fields.DateTimeField(db_column='ValidityFrom', blank=True, null=True)
     # validity_to = fields.DateTimeField(db_column='ValidityTo', blank=True, null=True)
-    #
     audit_user_id = models.IntegerField(db_column='AuditUserID', blank=True, null=True)
     # row_id = models.BinaryField(db_column='RowID', blank=True, null=True)
 
+    def __bool__(self):
+        return self.code is not None and len(self.code) >= 1
+
     def __str__(self):
         return self.code + " " + self.name
 
+    def __eq__(self, other):
+        equals = isinstance(other, Service) and \
+                 self.code == other.code and \
+                 self.name == other.name and \
+                 self.type == other.type and \
+                 self.level == other.level and \
+                 self.price == other.price and \
+                 self.care_type == other.care_type and \
+                 self.patient_category == other.patient_category and \
+                 self.frequency == other.frequency
+
+        if equals:
+            # optional string field -> making sure that None and empty string are treated as the same to avoid saving history
+            if bool(self.category) == bool(other.category):
+                if self.category:
+                    return self.category == other.category
+                else:
+                    return True
+
+        return False
+
+    # This method might raise problems with bulk delete using query sets
+    # https://docs.djangoproject.com/en/3.2/topics/db/models/#overriding-predefined-model-methods
+    def delete(self, hard_delete=False, *args, **kwargs):
```

Comment:
I'm surprised with the named param followed by positional *kwargs and named again. If it works, I wouldn't worry

Reply:
I'll be surpised too